src/FsAutoComplete/CodeFixes/ConvertCSharpLambdaToFSharpLambda.fs

```diff
@@ -0,0 +1,76 @@
+/// a codefix that rewrites C#-style '=>' lambdas to F#-style 'fun _ -> _' lambdas
+module FsAutoComplete.CodeFix.ConvertCSharpLambdaToFSharpLambda
+
+open FsToolkit.ErrorHandling
+open FsAutoComplete.CodeFix.Types
+open Ionide.LanguageServerProtocol.Types
+open FsAutoComplete
+open FsAutoComplete.LspHelpers
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Text
+
+let title = "Replace C#-style lambda with F# lambda"
+// adopted from `FSharp.Compiler.CodeAnalysis.FSharpParseFileResults.TryRangeOfParenEnclosingOpEqualsGreaterUsage`
+let private tryRangeOfParenEnclosingOpEqualsGreaterUsage input pos =
```

Comment:
we may be able to remove this patch now, as it seems to be available in FCS 41.0.3: https://www.fuget.org/packages/FSharp.Compiler.Service/41.0.3/lib/netstandard2.0/FSharp.Compiler.Service.dll/FSharp.Compiler.CodeAnalysis/FSharpParseFileResults?code=true#M%3AFSharp.Compiler.CodeAnalysis.FSharpParseFileResults.TryRangeOfParenEnclosingOpEqualsGreaterUsage%28FSharp.Compiler.Text.Position%29

Reply:
Both are slightly different from their `dotnet/fsharp` version.

Both `tryRangeOfParenEnclosingOpEqualsGreaterUsage` versions return something slightly different:
* `dotnet/fsharp` returns full range of `op_EqualsGreater` expression, lambda args range (lhs of =>) & body range (rhs of =>)
* this here returns args range & `=>` (operator) range

Difference:  
While `dotnet/fsharp` and prev FSAC replaced the whole lambda thing (including args & body text), I just add `fun ` and replace `->`  
-> less existing text to touch/replace  

And replacing full lambda body caused an issue with multi-line lambda bodys (-> see [test](https://github.com/Booksbaum/FsAutoComplete/blob/cc2a49f87da2f68b5eb19f17d8d924c3a0044b22/test/FsAutoComplete.Tests.Lsp/CodeFixTests.fs#L496-L527) -> joined line to single line (-> invalid F# code))


Edit: hm...this comment didn't commit? -> hopefully now

Reply:
ah, again you make perfect sense :)